setup.py

```diff
@@ -129,40 +133,87 @@ def run(self):
-        print("Unable to read requirements from the requirements.txt file"
-              "That indicates this copy of the source code is incomplete.")
-        sys.exit(2)
-
-# pycrypto or cryptography.   We choose a default but allow the user to
-# override it.  This translates into pip install of the sdist deciding what
-# package to install and also the runtime dependencies that pkg_resources
-# knows about
-crypto_backend = os.environ.get('ANSIBLE_CRYPTO_BACKEND', None)
-if crypto_backend:
-    if crypto_backend.strip() == 'pycrypto':
+def read_file(file_name):
+    """Read file and return its contents."""
+    with open(file_name, 'r') as f:
+        return f.read()
+
+
+def read_requirements(file_name):
+    """Read requirements file as a list."""
+    reqs = read_file(file_name).splitlines()
+    if not reqs:
+        raise RuntimeError(
+            "Unable to read requirements from the %s file"
+            "That indicates this copy of the source code is incomplete."
+            % file_name
+        )
+    return reqs
+
+
+PYCRYPTO_DIST = 'pycrypto'
+
+
+def get_crypto_req():
+    """Detect custom crypto from ANSIBLE_CRYPTO_BACKEND env var.
+
+    pycrypto or cryptography. We choose a default but allow the user to
+    override it. This translates into pip install of the sdist deciding what
+    package to install and also the runtime dependencies that pkg_resources
+    knows about.
+    """
+    crypto_backend = os.environ.get('ANSIBLE_CRYPTO_BACKEND', '').strip()
+
+    if crypto_backend == PYCRYPTO_DIST:
         # Attempt to set version requirements
-        crypto_backend = 'pycrypto >= 2.6'
+        return '%s >= 2.6' % PYCRYPTO_DIST
 
-    install_requirements = [r for r in install_requirements if not (r.lower().startswith('pycrypto') or r.lower().startswith('cryptography'))]
-    install_requirements.append(crypto_backend)
+    return crypto_backend or None
 
-# specify any extra requirements for installation
-extra_requirements = dict()
-extra_requirements_dir = 'packaging/requirements'
-for extra_requirements_filename in os.listdir(extra_requirements_dir):
-    filename_match = re.search(r'^requirements-(\w*).txt$', extra_requirements_filename)
-    if filename_match:
-        with open(os.path.join(extra_requirements_dir, extra_requirements_filename)) as extra_requirements_file:
-            extra_requirements[filename_match.group(1)] = extra_requirements_file.read().splitlines()
 
-# Retrieve the long description from the README
-with open('README.rst', 'r') as readme_file:
-    longdesc = readme_file.read()
+def append_crypto_to_req(req):
+    """Replace crypto requirements if customized."""
+    crypto_backend = get_crypto_req()
 
-setup(
+    if crypto_backend is None:
+        return req
+
+    def is_not_crypto(r):
+        CRYPTO_LIBS = PYCRYPTO_DIST, 'cryptography'
+        return not any(r.lower().startswith(c) for c in CRYPTO_LIBS)
+
+    return list(r for r in req if is_not_crypto(r)) + [crypto_backend]
```

Comment:
Nitpick, you can use a list comprehension here instead of calling list on a generator expression.

Reply:
oh, yeah. should be leftover from previous version.